lib/both.js

```diff
@@ -0,0 +1,185 @@
+    }
+    if (this._searchConfig.searchMode === 'global') {
+      return Meteor.call('__makeGlobalSearch', this._searchConfig, _onSearchComplete);
+    }
+  }
+
+  setMongoQuery(newMongoQuery) {
+    if (!_.isEqual(newMongoQuery, this._searchConfig.mongoQuery)) {
+      this._searchConfig.mongoQuery = newMongoQuery;
+      return this._updateQuery();
+    }
+  }
+
+  //May be used for infinite scroll or something like that
+  setLimit(newLimit) {
+    if (this._searchConfig.limit !== newLimit) {
+      this._searchConfig.limit = newLimit;
+      return this._updateQuery();
+    }
+  }
+
+  incrementLimit(step = 10) {
+    this._searchConfig.limit += step;
+    return this._updateQuery();
+  }
+
+  _getDataOfRealetiveCollection(subQuerys) {
+    let arr = [];
+    for (let query of subQuerys) {
+      let ids = this._realetiveCollection.find(query, {
+          fields: {_id: 1}
+        }).map(function(item){ 
+          return item._id; 
+        });
+      arr.push({[this._realetiveCollection._name]: {$in: ids} });
+    }
+    
+    return arr;
+  }
+
+  // Reactive data source
+  searchResult(options = {}) {
+    this._searchDep.depend();
+    if (this._searchConfig.searchMode === 'local') {
+      let querys = AntiSearchSource._buildSearchQuery(this._searchConfig);
+      let query = querys.searchQuery;
+      
+      if(querys.subQuerys) {
+        let subQuery = this._getDataOfRealetiveCollection(querys.subQuerys);
+        query = AntiSearchSource._addInSearchQuery(query, subQuery);
+      }
+      
+      _.extend(options, {
+        limit: this._searchConfig.limit,
+        transform: this._createTransformFn(options.transform)
+      });
+
+      return this._collection.find(query, options);
+    } else {
+      return this._searchResult;
+    }
+  }
+
+}
+
+
+this.AntiSearchSource = {
+  _allowRules: {},
+  _clientProto: AntiSearchSourceClient,
+  _escapeRegExpStr: function(str) {
+    return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
+  },
+  _buildSearchQuery: function(configEntry) {
+    let searchStringQueries = false;
+    let subQuerys = false;
+    if (configEntry.searchString) {
+      let escapedSearchString = this._escapeRegExpStr(configEntry.searchString);
+
+      searchStringQueries = this._buildQuerysByFields(configEntry.fields, escapedSearchString);
+      if(configEntry.realetive && configEntry.realetive.collection) {
+        subQuerys = this._buildQuerysByFields(configEntry.realetive.fields, escapedSearchString);
+      }
+    }
+
+    let searchQuery = {
+      $and: [_.extend({}, configEntry.mongoQuery)]
+    };
+
+    if (_.isArray(searchStringQueries)) {
+      searchQuery.$and.push({
+        $or: searchStringQueries
+      });
+    }
+
+    return {
+      searchQuery: searchQuery,
+      subQuerys: subQuerys
+    };
+  },
+  _buildQuerysByFields: function(fields, escapedSearchString) {
+    let result = fields.map(function(fieldName) {
+      return {[fieldName]: {
```

Comment:
hard to read. Multiline object should have only "{" at first line. All properties are listed from new line.